src/App.js

```diff
@@ -3,6 +3,7 @@ import './App.css';
 import FoodBox from './components/FoodBox';
 import foodsArr from './foods.json';
 import AddFood from "./components/AddFood";
+import Search from "./components/Search";
 
 
 function App() {
```

Comment:
In App.js you should have another state , to hold filteredArray and set initial value to original array -> that's the array you want to iterate over to display in broswer & send it to you child component to filter
```
  const [filteredFoods, setFilteredFoods] = useState(foods);
```

Now what's the reason for having additional state... We want to always be displaying 'filtered array'.

initially, it's value is the same original array from json / or database
when you filter once (have letters in search input field), we filter teh ORIGINAL array, and set new value of filtered array.
the second time you add letters to search, or remove letters -> we AGAIN filter from the ORIGINAL array. Because if were filtering from the filtered array furthermore (that's what you have at the moment, we are missign a lot of foods right?)